script.js

```diff
@@ -1,23 +1,37 @@
+var currentGameMode = "Please key in your name.";
+var userName = "";
+var numberOfWins = 0;
+var numberOfLose = 0;
+
 var main = function (input) {
-  if (input != "Scissors" && input != "Stone" && input != "Paper") {
-    return "Invalid input. Please select either 'Scissors', 'Paper', or 'Stone'.";
-  }
-  var myOutputValue = "The program won!";
-  var gameOutput = randomSPS();
-  console.log(gameOutput);
-  if (input == gameOutput) {
-    myOutputValue = "It is a draw!";
-  }
-  if (
-    (input == "Scissors" && gameOutput == "Paper") ||
-    (input == "Paper" && gameOutput == "Stone") ||
-    (input == "Stone" && gameOutput == "Scissors")
-  ) {
-    myOutputValue = "You won!";
+  var myOutputValue = "";
+  if (currentGameMode == "Please key in your name.") {
+    userName = input;
+    currentGameMode = "dice game";
+    myOutputValue = "Hello " + userName + "!";
+  } else if (currentGameMode == "dice game") {
+    if (input != "Scissors" && input != "Stone" && input != "Paper") {
+      return "Invalid input. Please select either 'Scissors', 'Paper', or 'Stone'.";
+    }
+    var gameOutput = randomSPS();
```

Comment:
Naming gameOutput as compOutput or compChoice might be a better name as it is clearer what the definition means (some might confuse gameOutput as being the result rather than the computers hand)